### DISCUSSION TITLE
Can't connect to wi-fi with NodeMCU-32S

### DISCUSSION THREAD

**Discussion by eapo** [micropython/micropython#15089] [ESP32]
### Checks

- [X] I agree to follow the MicroPython [Code of Conduct](https://github.com/micropython/micropython/blob/master/CODEOFCONDUCT.md) to ensure a safe and respectful space for everyone.

- [X] I've searched for [existing issues](https://github.com/micropython/micropython/issues) matching this bug, and didn't find any.

### Port, board and/or hardware

NodeMCU-32S

### MicroPython version

[ESP32_GENERIC-20240222-v1.22.2.bin](https://micropython.org/resources/firmware/ESP32_GENERIC-20240222-v1.22.2.bin)

### Reproduction

```python
import config  # Import configuration file
import network
import time
import uos

# Connect to WiFi using the configurations from config.py
wifi_ssid = config.wifi_ssid
wifi_password = config.wifi_password

wifi = network.WLAN(network.STA_IF)
wifi.active(True)
wifi.connect(wifi_ssid, wifi_password)

# Wait until connected
while not wifi.isconnected():
    time.sleep(1)

print("\033[32mConnected to Wi-Fi:\033[0m", wifi_ssid, wifi.ifconfig())
print("\033[32m")
```
> [eSP32-MicroPython/boot.py at main · eapo/eSP32-MicroPython](https://github.com/eapo/eSP32-MicroPython/blob/main/boot.py)

### Expected behaviour

Expected to print out the `wifi.ifconfig()`, but it is empty, and no new device appear in router log.

### Observed behaviour

While the same code running well on _NodeMCU-32 v1.2_ the _NodeMCU-32S_ it's not connecting. Can't figure out how to generate useful logs, because there is no any indication of any errors on console.

### Additional Information

Full code here: [eapo/eSP32-MicroPython: To control WS281X using DMX signals](https://github.com/eapo/eSP32-MicroPython/tree/main)

**Comment by robert-hh** [comment]
The Nodemcu devices are more a design specification than a specific board. various vendors sell devices under that name, and they all are Generic ESP32 devices. From what I can tell, the Generic ESP32 firmware is the right one. 
When you tell:

> Expected to print out the wifi.ifconfig(), but it is empty, and no new device appear in router log.

Does that mean that the `while ...` loop terminates?

**Reply by robert-hh** [reply]
Note: Converted to a discussion.

**Comment by shariltumin** [comment]
Please try the following script:

```python
import time
import network
import config

print(f'AP active: {network.WLAN(network.AP_IF).active(False)}')
wlan = network.WLAN(network.STA_IF)
print(f'STA active: {wlan.active(True)}')
wlan.disconnect()

ssid= config.wifi_ssid
pwd = config.wifi_password

cnt = 0
s   = 1

if ssid in [w[0].decode() for w in wlan.scan()]:
   wlan.connect(ssid, pwd)
   while not wlan.isconnected():
      print(f'status: {wlan.status()}')
      cnt+=1
      if cnt>10:
         print(f'give up after {cnt*s} seconds')
         break
      time.sleep(s)
   else:
      print('connected!')
      print(wlan.ifconfig())
else:
   print(f'SSID:{ssid} not found')
```

With a wrong password you will get something like:

```
AP active: False
STA active: True
status: 1000
status: 1000
status: 1000
status: 1000
status: 1001
status: 1001
status: 202
status: 202
status: 202
status: 1001
status: 1001
give up after 11 seconds
```
